js/walutomat.js

```diff
@@ -0,0 +1,272 @@
+                'INVALID ARGUMENT': ExchangeError,
+                'INVALID PARAMETER': ExchangeError,
+                'INSUFFICIENT_FUNDS': InsufficientFunds,
+                'MISSING_API_KEY': AuthenticationError,
+            },
+        });
+    }
+
+    fetchMarkets (params = {}) {
+        let currencies = ['EUR', 'GBP', 'USD', 'CHF', 'PLN'];
+        let markets = [];
+        let currenciesLength = currencies.length;
+        for (let base = 0; base < currenciesLength - 1; base++) {
+            let newIndex = base + 1;
+            for (let quote = newIndex; quote < currencies.length; quote++) {
+                markets.push ({
+                    'id': currencies[base] + '_' + currencies[quote],
+                    'symbol': currencies[base] + '/' + currencies[quote],
+                    'base': currencies[base],
+                    'quote': currencies[quote],
+                    'active': true,
+                    'precision': {
+                        'amount': 2,
+                        'price': 4,
+                    },
+                    'info': currencies[base] + ' - ' + currencies[quote] + ' exchange',
+                });
+            }
+        }
+        return markets;
+    }
+
+    async fetchBalance (params = {}) {
+        let balances = await this.privateGetAccountBalances ();
+        let result = {};
+        for (let i = 0; i < balances.length; i++) {
+            let balance = balances[i];
+            result[this.safeString (balance, 'currency')] = {
+                'free': this.safeFloat (balance, 'balanceAvailable'),
+                'used': this.safeFloat (balance, 'balanceReserved'),
+                'total': this.safeFloat (balance, 'balanceAll'),
+            };
+        }
+        return this.parseBalance (result);
+    }
+
+    async fetchOrderBook (symbol, limit = undefined, params = {}) {
+        let symbolObject = { 'symbol': this.formatSymbol (symbol) };
+        let response = await this.publicGetPublicMarketOrderbookSymbol (this.extend (symbolObject, params));
+        return this.parseOrderBook (response, undefined, 'bids', 'asks', 'price', 'baseVolume');
+    }
+
+    async fetchTrades (symbol, since = undefined, limit = undefined, params = {}) {
+        let request = {
+            'operationType': 'MARKET_FX',
+            'currencies': symbol,
+            'continueFrom': since,
+            'volume': limit,
+            'sortOrder': params['sortOrder'] || 'DESC',
+        };
+        let response = await this.privateGetAccountHistory (this.extend (request, params));
+        let result = [];
+        for (let i = 0; i < response.length; i++) {
+            result.push ({
+                'id': this.safeInteger (response, 'id'),
+            });
+        }
+        return result;
+    }
+
+    async createOrder (symbol, type, side, amount, price = undefined, params = {}) {
+        let submitId = params.submitId;
+        let pair = this.formatSymbol (symbol);
```

Comment:
Now, we need to remove all references to formatSymbol and parseSymbol, see how we usually work with markets here:

- https://github.com/ccxt/ccxt/wiki/Manual#symbols-and-market-ids
- https://github.com/ccxt/ccxt/blob/master/js/bittrex.js#L492
- https://github.com/ccxt/ccxt/blob/master/js/poloniex.js#L663

I'm sorry that this isn't really obvious, but if we simply use the plain parsing like you did by character substitution – that won't work with unified symbols, as those have to be recoded in some cases, so we use these types of parsing:

- https://github.com/ccxt/ccxt/blob/master/js/huobipro.js#L734
- https://github.com/ccxt/ccxt/blob/master/js/binance.js#L626

Reply:
Not sure whether it's good or not ;z I use now .marketId (to get from CCXT's symbol market ID, which is used by Walutomat) and .findSymbol (to get CCXT's symbol from market ID). If it OK ?